common/transaction/src/lib.rs

```diff
@@ -3,64 +3,124 @@
 multiversx_sc::imports!();
 multiversx_sc::derive_imports!();
 
-use eth_address::EthAddress;
-
 pub mod transaction_status;
 
 // revert protection
 pub const MIN_BLOCKS_FOR_FINALITY: u64 = 10;
 pub const TX_MULTIRESULT_NR_FIELDS: usize = 6;
 
+pub type GasLimit = u64;
 pub type TxNonce = u64;
 pub type BlockNonce = u64;
-pub type SenderAddressRaw<M> = ManagedBuffer<M>;
-pub type ReceiverAddressRaw<M> = ManagedBuffer<M>;
-pub type TxAsMultiValue<M> = MultiValue6<
+pub type SenderAddress<M> = ManagedAddress<M>;
+pub type ReceiverAddress<M> = ManagedAddress<M>;
+pub type TxAsMultiValue<M> = MultiValue7<
     BlockNonce,
     TxNonce,
-    SenderAddressRaw<M>,
-    ReceiverAddressRaw<M>,
-    TokenIdentifier<M>,
-    BigUint<M>,
+    SenderAddress<M>,
+    ReceiverAddress<M>,
+    ManagedVec<M, EsdtTokenPayment<M>>,
+    ManagedVec<M, Option<StolenFromFrameworkEsdtTokenData<M>>>,
+    Option<TransferData<M>>,
 >;
 pub type PaymentsVec<M> = ManagedVec<M, EsdtTokenPayment<M>>;
 pub type TxBatchSplitInFields<M> = MultiValue2<u64, MultiValueEncoded<M, TxAsMultiValue<M>>>;
 
 #[derive(TopEncode, TopDecode, NestedEncode, NestedDecode, TypeAbi, ManagedVecItem, Clone)]
-pub struct EthTransaction<M: ManagedTypeApi> {
-    pub from: EthAddress<M>,
-    pub to: ManagedAddress<M>,
-    pub token_id: TokenIdentifier<M>,
+pub struct TransferData<M: ManagedTypeApi> {
+    pub gas_limit: GasLimit,
+    pub function: ManagedBuffer<M>,
+    pub args: ManagedVec<M, ManagedBuffer<M>>,
+}
+
+#[derive(
+    TopDecode,
+    TopEncode,
+    NestedDecode,
+    NestedEncode,
+    Clone,
+    PartialEq,
+    Eq,
+    Debug,
+    TypeAbi,
+    ManagedVecItem,
+)]
+pub enum StolenFromFrameworkEsdtTokenType {
+    Fungible,
+    NonFungible,
+    SemiFungible,
+    Meta,
+    Invalid,
+}
+
+impl From<EsdtTokenType> for StolenFromFrameworkEsdtTokenType {
+    fn from(value: EsdtTokenType) -> Self {
+        match value {
+            EsdtTokenType::Fungible => StolenFromFrameworkEsdtTokenType::Fungible,
+            EsdtTokenType::NonFungible => StolenFromFrameworkEsdtTokenType::NonFungible,
+            EsdtTokenType::SemiFungible => StolenFromFrameworkEsdtTokenType::SemiFungible,
+            EsdtTokenType::Meta => StolenFromFrameworkEsdtTokenType::Meta,
+            EsdtTokenType::Invalid => StolenFromFrameworkEsdtTokenType::Invalid,
+        }
+    }
+}
+
+#[derive(
+    TopDecode, TopEncode, NestedDecode, NestedEncode, TypeAbi, Debug, ManagedVecItem, Clone,
+)]
+pub struct StolenFromFrameworkEsdtTokenData<M: ManagedTypeApi> {
```

Comment:
Can you rename `StolenFromFrameworkEsdtTokenData`?

Reply:
It will be introduced in framework in the next version. Will be deleted then :)